tests/test_compose_curated_destination.py

```
@@ -0,0 +1,538 @@
+                        root / "curated",
+                        units_migration=calibration,
+                    )
+                self.assertFalse((root / "curated").exists())
+
+    def test_refuses_unsafe_destinations(self):
+        with tempfile.TemporaryDirectory() as td:
+            root = Path(td)
+            source = build_source_run(root / "run")
+            compose_curated.compose_run(source, root / "curated")
+
+            with self.assertRaises(compose_curated.ComposeError):
```

Ruff (0.16.3)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)